src/core/tUnification.ml

```diff
@@ -989,6 +1017,7 @@ and with_variance uctx f t1 t2 =
 		raise (Unify_error l)
 
 and unify_with_access uctx f1 t1 f2 =
+	let uctx = get_nested_context uctx in
```

Comment:
This doesn't seem right to me. Why should `unify_with_access`, which is used to compare class field types, always allow abstract casts?

Reply:
Well because right now they are always allowed, no? (so `allow_abstract_cast` is a tracker and not an option that you supposed to set outside)

If there is a desire to have an ability to disallow abstract casts specifically, then `allow_abstract_cast` will be `bool option`: with `Some(true)` as default and `None` meaning disallowing.

Should it be updated so?

Reply:
Well, right, maybe I'm not asking the right question here. What I'm wondering about is what makes `unify_with_access` so special that it's the only place where we use `get_nested_context` and thus `allow_abstract_cast = true`.

Reply:
There are two places were we change type dimensions. First is when we look into fields of a structure for structural sub-typing, second is about type parameters comparison.

Right now type parameters use only `type_eq` calls, no `unify` ones, so they do not care about `allow_abstract_cast`  value, thus no need for reset. `get_nested_context` can be added to `unify_type_params` just for consistency/future sake if desired.

Is there some other place where reset should be made?

Reply:
What about function argument/return type unification? In my mind that's on the same level as structure field types.

Reply:
Right, done.